lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -829,10 +835,14 @@ class ParseableInterfaceModuleLoaderImpl {
     if (fs.exists(cachedOutputPath)) {
       auto buf = fs.getBufferForFile(cachedOutputPath);
 
-      // Unable to open the cached module, treat it as if it wasn't there
-      // and rebuild.
-      if (!buf)
-        return notFoundError;
+      // If we're unable to read this corrupted file, delete it from the
+      // cache and try to load again.
+      if (!buf) {
+        llvm::sys::fs::remove(cachedOutputPath);
```

Comment:
Probably for later, but per discussion yesterday: what do we do if this delete fails?

Reply:
I don't know if there's anything we can do except error out and ask the user to delete the file 😥

That Should Never Happen™, because ideally we're the only process that touches this file, but...

Reply:
The main case I'm thinking of is someone compiling with `sudo` For Some Reason and yet using the same module cache.

Reply:
I've added a FIXME -- I'll make it diagnose in a follow-up